Makefile

```diff
@@ -0,0 +1,50 @@
+.PHONY: all clean generate fmt
+.PHONY: tidy get build test up
+
+GO ?= go
+GOFMT ?= gofmt
+GOFMT_FLAGS = -w -l -s
+GOGENERATE_FLAGS = -v
+
+GOBIN ?= $(GOPATH)/bin
+
+REVIVE ?= $(GOBIN)/revive
+REVIVE_FLAGS ?= -formatter friendly
+REVIVE_INSTALL_URL ?= github.com/mgechev/revive
+
+TMPDIR ?= .tmp
+
+all: get generate fmt tidy build
+
+clean:
+	rm -rf $(TMPDIR)
+
+fmt:
+	@find . -name '*.go' | xargs -r $(GOFMT) $(GOFMT_FLAGS)
+
+tidy: fmt $(REVIVE)
+	$(GO) mod tidy
+	$(GO) vet ./...
+	$(REVIVE) $(REVIVE_RUN_ARGS) ./...
+
+get:
+	$(GO) get -v ./...
+
+build:
+	$(GO) build -v ./...
+
+test:
+	$(GO) test -v ./...
+
+up:
+	$(GO) get -u -v ./...
+	$(GO) mod tidy
+	$(GO) install -v $(REVIVE_INSTALL_URL)
+
+generate:
+	@git grep -l '^//go:generate' | sed -n -e 's|\(.*\)/[^/]\+\.go$$|\1|p' | sort -u | while read d; do \
+		git grep -l '^//go:generate' "$$d"/*.go | xargs -r $(GO) generate $(GOGENERATE_FLAGS); \
```

Comment:
Why not go generate ./...?

Reply:
I honestly don't remember and can't justify it. I've been carrying it around from other projects without asking myself if it was relevant. I've replaced it with a single go generate call